docs/docs/reference-react-component.md

```diff
@@ -44,27 +99,47 @@ constructor(props) {
 
```

Comment:
>It's okay to initialize state based on props.

Only if you know what you're doing. Most beginners do this by mistake. This effectively "forks" the props and can lead to bugs.

Instead of syncing props to state, you often want to lift the state up: https://github.com/facebook/react/pull/7920#issuecomment-252497085.

If you "fork" props by using them for state, you might also want to implement `componentWillReceiveProps()` to keep the state up-to-date with them. But lifting state up is often easier and less bug-prone.